document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -1254,6 +1258,130 @@ private String getSubDocUpdateQuery() {
         tableIdentifier, DOCUMENT, DOCUMENT, ID);
   }
 
+  static class PostgresResultIteratorWithBasicTypes extends PostgresResultIterator {
+
+    public PostgresResultIteratorWithBasicTypes(ResultSet resultSet) {
+      super(resultSet);
+    }
+
+    @Override
+    public Document next() {
+      try {
+        if (!cursorMovedForward) {
+          resultSet.next();
+        }
+        // reset the cursorMovedForward state, if it was forwarded in hasNext.
+        cursorMovedForward = false;
+        return prepareDocument();
+      } catch (IOException | SQLException e) {
+        System.out.println("prepare document failed!");
+        closeResultSet();
+        return JSONDocument.errorDocument(e.getMessage());
+      }
+    }
+
+    protected Document prepareDocument() throws SQLException, IOException {
+      ObjectNode jsonNode = MAPPER.createObjectNode();
+
+      // Get metadata to iterate through all columns
+      ResultSetMetaData metaData = resultSet.getMetaData();
+      int columnCount = metaData.getColumnCount();
+
+      for (int i = 1; i <= columnCount; i++) {
+        String columnName = metaData.getColumnName(i);
+        String columnType = metaData.getColumnTypeName(i);
+
+        addColumnToJsonNode(jsonNode, columnName, columnType, i);
+      }
+
+      // Remove document ID if needed
+      if (shouldRemoveDocumentId()) {
+        jsonNode.remove(DOCUMENT_ID);
+      }
+
+      return new JSONDocument(MAPPER.writeValueAsString(jsonNode));
+    }
+
+    private void addColumnToJsonNode(
+        ObjectNode jsonNode, String columnName, String columnType, int columnIndex)
+        throws SQLException {
+      switch (columnType.toLowerCase()) {
```

Comment:
Did we face some issue, while using this one - mapValueToJsonNode? Can that be reused?